modin/pandas/base.py

```diff
@@ -222,10 +222,7 @@ def _validate_other(
         self,
         other,
         axis,
-        numeric_only=False,
-        numeric_or_time_only=False,
-        numeric_or_object_only=False,
-        comparison_dtypes_only=False,
+        type_check=False,
```

Comment:
I wonder if we should name this `dtype_check` (extra `d`) to highlight this is about data types?..